src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -565,5 +544,95 @@ internal void TakeSnapshot(bool outputDelta)
         {
             throw new NotSupportedException($"{methodName} is not supported for this metric type.");
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void HistogramUpdate(double number, bool hasBuckets, bool hasMinMax)
```

Comment:
I think we should investigate having a single method that would account for all scenarios vs having a dedicated method for each of them (without these extra if-checks)

The `Update` and `Snapshot` methods for Histogram use a spin-lock mechanism. We want the time taken by these methods during the lock to be as less as possible. Moreover, `Update` method is in the hot path.

We need Stress Test results for before and after this change.

Reply:
Added this perf improvement to the following issue: https://github.com/open-telemetry/opentelemetry-dotnet/issues/3428

Reply:
this might be a minor regression from existing scenario where configured just SUM and Count. Can you run the main vs this branch with MinMax=false, and share numbers?
I'll leave this comment open, but wont block PR merge as we can still address perf regression, if any, as follow ups

Reply:
I'll just unfold these functions 👍